casper/src/main/resources/PoS.rhox

```diff
@@ -344,17 +405,29 @@ new  PoS,
           }
         } |
 
-        contract deposit (@fromUserPk, @amount, returnCh) = {
+        contract deposit (@fromUserPk, @amount, @activeValidators, returnCh) = {
           new vaultCh,
               revAddressCh,
+              indexSender,
+              getBlockData(`rho:block:data`), blockDataCh,
               authKeyCh in {
             @RevVault!("deployerAuthKey", *authKeyCh) |
             for (@authKey <- authKeyCh) {
               revAddressOps!("fromPublicKey", fromUserPk, *revAddressCh) |
+              getBlockData!(*blockDataCh) |
               for (@fromRevAddress <- revAddressCh) {
                 @RevVault!("findOrCreate", fromRevAddress, *vaultCh) |
-                for (@(true, fromVault) <- vaultCh) {
-                  @fromVault!("transfer", posRevAddress, amount, authKey, *returnCh)
+                for (@(true, fromVault) <- vaultCh; _, _, @sender <- blockDataCh) {
+                  @ListOps!("indexOf", activeValidators, sender, *indexSender) |
+                  for (@idx <- indexSender) {
+                    for (@vaults <<- perActiveValidatorPosVaultsCh) {
+                      match vaults.nth(idx).nth(1) {
+                        activeValidatorVault => {
+                          @fromVault!("transfer", activeValidatorVault, amount, authKey, *returnCh)
+                        }
+                      }
```

Comment:
this looks better as it makes it obvious that the second nth is applied on a tuple

```scala
                     match vaults.nth(idx) {
                        (activeValidatorVault, _) => {
                          @fromVault!("transfer", activeValidatorVault, amount, authKey, *returnCh)
                        }
                      }
```